tools/samtools.wdl

```diff
@@ -1006,3 +1018,224 @@ task fixmate {
         maxRetries: 1
     }
 }
+
+task position_sorted_fixmate {
```

Comment:
I hesitate to add this to the repo as a task. I'm guessing that the intent is to avoid creating a number of extra intermediate outputs, but it seems like this should be a workflow that can be included as a subworkflow to get this behavior.

Reply:
Is the fear that this opens the door for having a name-sorted **and** position-sorted version of each task? Because I had not thought of that, and it is a scary idea.

> I'm guessing that the intent is to avoid creating a number of extra intermediate outputs,

Yes. Specifically I wanted to avoid putting a collated BAM on disk. I don't have the numbers in front of me, but collation can cause some astonishingly huge BAMs in the worst case. (And no, I have not figured out how to predict what causes the worst case.)

Flashback to when we first swapped from running `picard SamToFastq` to running `samtools collate` then `samtools fastq` **as separate tasks**. Having collated WGS BAMs for each sample hogged _all_ the disk space wherever I was running QC and the LSF team was upset.

My point is that I think we need to think carefully about how to prevent massive intermediate file build-up.

>  it seems like this should be a workflow that can be included as a subworkflow to get this behavior.

Can you elaborate on this idea? I'm not following, unless you're suggesting we just eat the intermediate file cost.

Good news: we don't need this task. I kept us using `picard MarkDuplicates`, which means this is not called by any of our workflows. We could just delete it. I'm not convinced that's the _best_ choice, but it is a choice.

🤔 🤔 🤔